client/plasma_cash/utils.go

```diff
@@ -0,0 +1,37 @@
+// +build evm
+
+package plasma_cash
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/ethereum/go-ethereum/crypto/sha3"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/loomnetwork/go-loom/common/evmcompat"
+)
+
+func Sha3WithRlpEncoding(value interface{}) ([]byte, error) {
+	encodedBytes, err := rlp.EncodeToBytes(value)
+	if err != nil {
+		return nil, err
+	}
+
+	return Sha3(encodedBytes), nil
+}
+
+func Sha3(input []byte) []byte {
+	hasher := sha3.NewKeccak256()
+	hasher.Write(input)
+	return hasher.Sum(nil)
+}
+
+func SolidityTypedSign(hash []byte, key *ecdsa.PrivateKey) ([]byte, error) {
```

Comment:
I don't know why any of this is here, we already have this functionality in go-loom, e.g.
```golang
hash = ssha.SoliditySHA3(
	ssha.Address(common.HexToAddress(prevOwner)),
	ssha.Uint256(new(big.Int).SetUint64(senderNonce)),
	txHash, // hash of PlasmaTx
)
sig, err := evmcompat.GenerateTypedSig(hash, senderPrivateKey, evmcompat.SignatureType_EIP712)
if err != nil {
	return err
}
```